code/modules/shuttle/shuttle.dm

```diff
@@ -184,6 +184,13 @@
 		SSshuttle.stationary -= src
 	. = ..()
 
+/obj/docking_port/stationary/Moved(atom/oldloc, dir, forced)
+	. = ..()
+	if(area_type) // We already have one
```

Comment:
Maybe if you move the dock, by advanced shuttle console, better to always get new area or check and set it if different.
In case of setting it in the drydock, lavalend or another similar thing.

Now we can not put it in lavalend using custom methods, but it is now.

Reply:
I purposefully didn't handle that because if it's anything other than null it may be a custom area being used. 

The advanced shuttle console needs to handle its own dock if it wants to move around a stationary dock which was never designed to be moved in the first place really. This just makes creating in nullspace before moving to a spot supported.

Reply:
https://github.com/tgstation/tgstation/blob/master/code/modules/shuttle/navigation_computer.dm#L146

Reply:
Yes I know

Reply:
This i mean. The part of the zone replaced by space
![image](https://user-images.githubusercontent.com/7734424/59932522-072a0680-9450-11e9-899b-4f27474aca9d.png)
![image](https://user-images.githubusercontent.com/7734424/59932788-a4853a80-9450-11e9-977c-a9b39bf60a3e.png)